drivers/wallclock/sun6i-cnt64.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <error.h>
+#include <mmio.h>
+#include <drivers/clock.h>
+#include <drivers/wallclock.h>
+#include <util.h>
+
+#define  OSC24M_CNT64_CTRL_REG		0x28
+#define OSC24M_CNT64_LOW_REG		0x284
+#define OSC24M_CNT64_HIGH__REG		0x288
+
+static uint64_t
+sunxi_wallclock_get_time(struct device *dev)
+{
+	uint64_t time;
+	uint32_t high_reg;
+	uint32_t low_reg
+
+	mmio_setbits32(dev->regs + OSC24M_CNT64_CTRL_REG, BIT(1);
+
+	while(mmio_read32(dev->regs + OSC24M_CNT64_CTRL_REG) ){
```

Comment:
There are several bits in this register. You need to test for the specific bit (bit 1), and you have the condition backwards. From the manual:

> 1: to latch the 64-bit Counter to the Low/Hi registers and it will change to zero after the registers are latched.

This means you have to wait until the bit goes back to zero to read the HIGH/LOW registers.